src/link/languages.js

```diff
@@ -340,5 +342,85 @@ const languagesWithWikis = [
 	'mus',
 ];
 
-export const isLanguageWithWiki = ( language ) =>
+const limit = defaultLanguages.length;
+const languages = getLanguages();
+
+const isLanguageWithWiki = ( language ) =>
 	languagesWithWikis.indexOf( language ) !== -1;
+
+const normalize = ( result, language ) => {
+	if ( isLanguageWithWiki( language ) ) {
+		result.push( {
+			name: languages[ language ][ 2 ],
+			code: language,
+		} );
+	} else if ( language === 'en-simple' ) {
+		result.push( {
+			name: languages[ language ][ 2 ],
+			code: 'simple',
+		} );
+	}
+	return result;
+};
+
+const getLocalized = ( language, userLang ) => {
+	const localizedName = new Intl.DisplayNames( [ userLang ], {
+		type: 'language',
+	} );
+	const invalid = [
+		'bat-smg',
+		'be-x-old',
+		'cbk-zam',
+		'fiu-vro',
+		'map-bms',
+		'roa-rup',
+		'zh-classical',
+		'zh-min-nan',
+		'zh-yue',
+		'zh-cdo',
+	];
+
+	if ( invalid.indexOf( language ) === -1 ) {
+		return localizedName.of( language ).toLowerCase();
+	}
+	return false;
+};
+
+export const filterLanguages = ( target, lang, setValue, setItems ) => {
```

Comment:
Instead of passing `setValue` and `setItems`, could you set the value before calling this function, return the items and set those after calling this function? It's uncommon to have a util set state back into a component.

Reply:
Yea of course makes sense